File: interviews/leetcode/leetcode-75/bit-manipulation/136. Single Number/solution.py
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        res = 0
        # XOR all elements in the array
        for num in nums:
            res ^= num
        return res

# Main function (commented out by default)
# def main():
#     solution = Solution()
#     nums = [4, 1, 2, 1, 2]
#     print(solution.singleNumber(nums))  # Output: 4

File: interviews/leetcode/leetcode-75/bit-manipulation/136. Single Number/solution.go
package main

// Function to find the single number
func singleNumber(nums []int) int {
	res := 0
	// XOR all elements in the array
	for _, num := range nums {
		res ^= num
	}
	return res
}

// Main function (commented out by default)
// func main() {
//     nums := []int{4, 1, 2, 1, 2}
//     fmt.Println(singleNumber(nums)) // Output: 4
// }
